### PR TITLE
CI: Use Qt 5 and GCC 6 on Ubuntu CI builds

### PR DESCRIPTION
**Pull request by lioncash**
The OSX build already uses Qt 5, so nothing needs to change here for that.

This also bumps the distribution of Ubuntu that we use up from Precise Pangolin (12.04) to Trusty Tahr (14.04). We also finally drop support for Qt 4 variants with this.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -201,16 +200,8 @@ if (ENABLE_QT)
...
-    if (NOT CITRA_FORCE_QT4)
-        find_package(Qt5 COMPONENTS Widgets OpenGL ${QT_PREFIX_HINT})
-        set(CITRA_QT_LIBS Qt5::Widgets Qt5::OpenGL)
-    endif()
-
-    if (CITRA_FORCE_QT4 OR NOT Qt5_FOUND)
-        # Try to fallback to Qt4
-        find_package(Qt4 REQUIRED COMPONENTS QtGui QtOpenGL ${QT_PREFIX_HINT})
-        set(CITRA_QT_LIBS Qt4::QtGui Qt4::QtOpenGL)
-    endif()
```
Comment (linkmauve) [inline CMakeLists.txt]: This should probably be in its own commit, with a more explicit summary, since some users might still be using that fallback.

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,26 @@
-os:
-  - linux
-  - osx
-
 language: cpp
 
+matrix:
+  include:
+    - os: linux
+      sudo: required
```
Comment (linkmauve) [inline .travis.yml]: What does it require sudo for?  I don’t see any change here that would cause that change.

**Comment by linkmauve** [comment]
Aside from that, LGTM!

File: .travis.yml (hunk 2 above)
Comment (lioncash) [inline .travis.yml]: That was a leftover from following Travis' guide for setting up the build matrix. I removed it in the revised commits.

File: CMakeLists.txt (hunk 1 above)
Comment (lioncash) [inline CMakeLists.txt]: Done

**Comment by bunnei** [comment]
Changes LGTM... What's with the apt-get install failed on Travis?

File: .travis.yml (hunk 2 above)
Comment (lioncash) [inline .travis.yml]: Actually nevermind this is required in order for some packages to be found.

**Comment by lioncash** [comment]
@bunnei all good now.

File: .travis.yml [hunk 3]
```diff
@@ -1,22 +1,26 @@
-os:
-  - linux
-  - osx
-
 language: cpp
 
+matrix:
+  include:
+    - os: linux
+      sudo: true
```
Comment (yuriks) [inline .travis.yml]: Why do you need this? This puts us in Travis VMs instead of containers (at least last time I checked) which slows down builds.

Comment (jroweboy) [inline .travis.yml]: citra is currently using the `trusty` beta environment (Ubuntu 14.04) which means you can't use the containers since they are based on ubuntu 12.04 still. Whether we need trusty is still up for debate. I don't know for sure

See https://docs.travis-ci.com/user/ci-environment/ for more info
